test/unit/specs/utils/dom/awaitSelector.spec.js

```diff
@@ -31,39 +31,40 @@ describe("DOM::awaitSelector", () => {
     removeNode(nodes[0]);
   };
 
-  const awaitSelectorAndAssert = (id, win, doc, done) => {
+  const awaitSelectorAndAssert = (id, win, doc) => {
     const result = awaitSelector(`#${id}`, selectNodes, 1000, win, doc);
 
     createAndAppendNodeDelayed(id);
 
-    result
+    return result
       .then(nodes => {
-        done();
-        cleanUp(id);
         expect(nodes[0].tagName).toEqual("DIV");
       })
-      .catch(() => {
-        done();
+      .finally(() => {
         cleanUp(id);
-        fail(`${id} should be found`);
+      })
+      .catch(() => {
+        throw new Error(`${id} should be found`);
       });
   };
 
-  it("await via MutationObserver", done => {
-    awaitSelectorAndAssert("abc", window, document, done);
+  it("await via MutationObserver", () => {
+    return awaitSelectorAndAssert("abc", window, document);
   });
 
-  it("await via requestAnimationFrame", done => {
-    const win = { requestAnimationFrame: window.requestAnimationFrame };
+  it("await via requestAnimationFrame", () => {
+    const win = {
+      requestAnimationFrame: window.requestAnimationFrame.bind(window)
```

Comment:
This was also a reason for failure. `requestAnimationFrame` wasn't working as expected without it being bound to window.